bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/OnOffValue.java

```diff
@@ -44,55 +43,40 @@ public class OnOffValue extends Value {
-     * @param offState The OFF value string. This will be compared to MQTT messages. Defaults to offComamand if null, or
-     *            "OFF" if both are null.
-     * @param onCommand The ON value string. This will be send in MQTT messages. Defaults to onState if null, or "ON" if
-     *            both are null.
-     * @param offCommand The OFF value string. This will be send in MQTT messages. Defaults to offCommand if null, or
-     *            "OFF" if both are null.
+     * @param onStates The list of ON value strings. These will be compared to MQTT messages. Defaults to "ON" if null.
+     * @param offStates The list of OFF value strings. These will be compared to MQTT messages. Defaults to "OFF" if
+     *            null.
+     * @param onCommand The ON value string. This will be sent in MQTT messages. Defaults to the first onState if null.
+     * @param offCommand The OFF value string. This will be sent in MQTT messages. Defaults to the first offState if
+     *            null.
      */
-    public OnOffValue(@Nullable String onState, @Nullable String offState, @Nullable String onCommand,
+    public OnOffValue(@Nullable List<String> onStates, @Nullable List<String> offStates, @Nullable String onCommand,
             @Nullable String offCommand) {
-        this(new String[] { defaultArgument(onState, onCommand, OnOffType.ON.name()) },
-                new String[] { defaultArgument(offState, offCommand, OnOffType.OFF.name()) },
-                defaultArgument(onCommand, onState, OnOffType.ON.name()),
-                defaultArgument(offCommand, offState, OnOffType.OFF.name()));
-    }
-
-    /**
-     * Creates a new SWITCH On/Off value.
-     *
-     * @param onStates A list of valid ON value strings. This will be compared to MQTT messages.
-     * @param offStates A list of valid OFF value strings. This will be compared to MQTT messages.
-     * @param onCommand The ON value string. This will be send in MQTT messages.
-     * @param offCommand The OFF value string. This will be send in MQTT messages.
-     */
-    public OnOffValue(String[] onStates, String[] offStates, String onCommand, String offCommand) {
         super(CoreItemFactory.SWITCH, List.of(OnOffType.class, StringType.class));
-        this.onStates = Stream.of(onStates).filter(not(String::isBlank)).collect(Collectors.toSet());
-        this.offStates = Stream.of(offStates).filter(not(String::isBlank)).collect(Collectors.toSet());
-        this.onCommand = onCommand;
-        this.offCommand = offCommand;
+        this.onStates = onStates == null ? Set.of(OnOffType.ON.name())
+                : onStates.stream().filter(not(String::isBlank)).collect(Collectors.toSet());
+        this.offStates = offStates == null ? Set.of(OnOffType.ON.name())
+                : offStates.stream().filter(not(String::isBlank)).collect(Collectors.toSet());
+        this.onCommand = onCommand == null ? (onStates == null ? OnOffType.ON.name() : onStates.get(0)) : onCommand;
```

Comment:
I'm wondering whether onStates could be not null, but having size() == 0, in which case, get(0) will cause index out of bounds.

Reply:
Could be possible, true.